src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/index.tsx

```diff
@@ -122,7 +132,80 @@ class IssueRuleEditor extends AsyncView<Props, State> {
     return endpoints as [string, string][];
   }
 
-  handleSubmit = async () => {
+  pollHandler = async (quitTime: number) => {
+    if (Date.now() > quitTime) {
+      addErrorMessage(t('Looking for that channel took too long :('));
+      this.setState({loading: false});
+      return;
+    }
+
+    const {organization, project} = this.props;
+    const {uuid} = this.state;
+    const origRule = {...this.state.rule};
+    let response: RuleTaskResponse = null;
+
+    try {
+      response = await this.api.requestPromise(
```

Comment:
@MeredithAnya I realize the original code probably looked like this but once I saw `RuleTaskResponse` sometimes being typed as null, I realized there's a better way. I would just move everything in the `if (response) {` so inside the try/catch so you can do `const response RuleTaskResponse  =`. That way, you don't have to have response also be of type `null`.